debezium-core/src/test/java/io/debezium/transforms/ByLogicalTableRouterTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms;
+
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Mario Mueller
+ */
+public class ByLogicalTableRouterTest {
+
+    @Test(expected = ConnectException.class)
+    public void testBrokenKeyReplacementConfiguration() {
+        final ByLogicalTableRouter<SourceRecord> subject = new ByLogicalTableRouter<>();
+        final Map<String, String> props = new HashMap<>();
+
+        props.put("topic.regex", "someValidRegex(.*)");
+        props.put("topic.replacement", "$1");
+        props.put("key.field.regex", "If this is set, key.field.replacement must be non-null");
+        subject.configure(props);
+    }
+
+    @Test(expected = ConnectException.class)
+    public void testBrokenTopicReplacementConfiguration() {
+        final ByLogicalTableRouter<SourceRecord> subject = new ByLogicalTableRouter<>();
+        final Map<String, String> props = new HashMap<>();
+
+        props.put("topic.regex", "someValidRegex(.*)");
+        subject.configure(props);
+    }
+
+    // FIXME: This SMT can use more tests for more detailed coverage.
+    // The creation of a SourceRecord is required for each test
```

Comment:
Right, but there is yet no convenient way to create a `SourceRecord` in the same manner as DBZ emits it, right? The transformation is very much based on the structure of DBZ and therefore a normal `SourceRecord` won't do the trick in the test.